src/Compilers/Core/Portable/NativePdbWriter/PdbWriter.cs

```diff
@@ -742,18 +746,19 @@ public void EmbedSourceLink(Stream stream)
         }
 
         /// <summary>
-        /// Write document entries for any embedded text document that does not yet have an entry.
+        /// Write document entries for all debug documents that do not yet have an entry.
         /// </summary>
         /// <remarks>
         /// This is done after serializing method debug info to ensure that we embed all requested
         /// text even if there are no corresponding sequence points.
         /// </remarks>
-        public void WriteRemainingEmbeddedDocuments(IEnumerable<DebugSourceDocument> embeddedDocuments)
+        public void WriteRemainingDebugDocuments(IReadOnlyDictionary<string, DebugSourceDocument> documents)
         {
-            foreach (var document in embeddedDocuments)
+            foreach (var kvp in documents
+                .Where(kvp => !_documentIndex.ContainsKey(kvp.Value))
+                .OrderBy(kvp => kvp.Key))
```

Comment:
>OrderBy [](start = 17, length = 7)

I didn't understand why we need ordering. There are other methods that invoke `AddDocumentIndex` and don't seem to do so in any particular order (namely `GetDocumentIndex`).

Reply:
This is to resolve https://github.com/dotnet/roslyn/pull/39136#discussion_r333275492
I think there is some kind of ordering in documents for ones having sequence points likely governed by sequence points ordering. And we want to have some consistence for the order of documents without sequence points. Let them be ordered by file names. @tmat may want to comment more.

Reply:
Any time we produce output based on dictionary enumeration we need to order the results in order to achieve determinism.

Reply:
Makes sense. Thanks